hw/net/rocker/rocker_hw.h

```diff
@@ -146,6 +147,7 @@ typedef struct rocker_tlv {
 enum {
     ROCKER_TLV_CMD_UNSPEC,
     ROCKER_TLV_CMD_TYPE,                /* u16 */
+    ROCKER_TLV_CMD_WORLD,               /* u32 - world id */
```

Comment:
Yes, currently the cmds are unique across the worlds (but don't have to
be). I started without the world_id tlv but introduced the world_id tlv for
the following two reasons. I also took the cue from the comment just above
the switch in cmd_consume() function. -

1. When a new cmd for any world is added, I had to add it in two places the
switch in cmd_consume() function in rocker.c and again in world-specific
command handler (e.g of_dpa_cmd() or rocker_p4_rmt_cmd())
2. As we add more P4 pipelines (worlds), we plan to provide an uniform
interface for all table operations (CRUD). So new table specific TLVs are
not visible in rocker_rmt_cmd either).

Basically, when something changes in a world (pipeline), I wanted to
minimize code changes to the common code in rocker.c and rocker_p4_rmt.c
(this file is common to different P4 worlds). With world_id tlv, I removed
changes in rocker.c and with a table_op blob, I do not need to change the
rocer_p4_rmt.c
Also as we add more worlds they can have there own TLV space (like nested
TLVs, but w/o adding another level of nesting).

On Sun, Apr 19, 2015 at 9:03 PM, Scott Feldman <notifications@github.com>
wrote:

> In hw/net/rocker/rocker_hw.h
> <https://github.com/pbhide/qemu-rocker/pull/1#discussion_r28663018>:
>
> > @@ -146,6 +147,7 @@ typedef struct rocker_tlv {
> >  enum {
> >      ROCKER_TLV_CMD_UNSPEC,
> >      ROCKER_TLV_CMD_TYPE,                /* u16 */
> > +    ROCKER_TLV_CMD_WORLD,               /* u32 - world id */
>
> Since the ROCKER_TLV_CMD_TYPE_P4_xxx cmds are already only for P4 world,
> it seems we don't need this ROCKER_TLV_CMD_WORLD. In other words, the world
> is encoded in the cmd type.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pbhide/qemu-rocker/pull/1/files#r28663018>.
>